tool/tsh/putty_config_windows.go

```diff
@@ -0,0 +1,313 @@
+func formatLocalCommandString(tshPath string, cluster string) string {
+	// replace the placeholder "%cluster" with the actual cluster name as passed to the function
+	clusterString := strings.ReplaceAll(puttyProxyTelnetCommand, "%cluster", cluster)
+	// PuTTY needs its paths to be double-escaped i.e. C:\\Users\\User\\tsh.exe
+	escapedTshPath := strings.ReplaceAll(tshPath, `\`, `\\`)
+	return strings.ReplaceAll(clusterString, "%tsh", escapedTshPath)
+}
+
+// onPuttyConfig handles the `tsh config putty` subcommand
+func onPuttyConfig(cf *CLIConf) error {
+	if runtime.GOOS != constants.WindowsOS {
+		return trace.NotImplemented("PuTTY config is only implemented on Windows")
+	}
+
+	tc, err := makeClient(cf, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// connect to proxy to fetch cluster info
+	proxyClient, err := tc.ConnectToProxy(cf.Context)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer proxyClient.Close()
+
+	// parse out proxy details
+	proxyHost, _, err := net.SplitHostPort(tc.Config.SSHProxyAddr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// get the public key of the Teleport host CA so we can add it to PuTTY's list of trusted keys
+	var hostCAKnownHostsLine string
+	err = tc.WithRootClusterClient(cf.Context, func(clt auth.ClientI) error {
+		hostCAKnownHostsLine, err = client.ExportAuthorities(cf.Context, clt, client.ExportAuthoritiesRequest{AuthType: "host"})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+	})
+	_, _, hostCAPublicKeyBytes, _, _, err := ssh.ParseKnownHosts([]byte(hostCAKnownHostsLine))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	hostCAPublicKey := strings.TrimSuffix(string(ssh.MarshalAuthorizedKey(hostCAPublicKeyBytes)), "\n")
+
+	// get root cluster name
+	rootClusterName, rootErr := proxyClient.RootClusterName(cf.Context)
+	// TODO(gus): figure out what to do (if anything) about leaf clusters
+	//leafClusters, leafErr := proxyClient.GetLeafClusters(cf.Context)
+	_, leafErr := proxyClient.GetLeafClusters(cf.Context)
+	if err := trace.NewAggregate(rootErr, leafErr); err != nil {
+		return trace.Wrap(err)
+	}
+
+	keysDir := profile.FullProfilePath(tc.Config.KeysDir)
+	ppkFilePath := keypaths.PPKFilePath(keysDir, proxyHost, tc.Config.Username)
+	certificateFilePath := keypaths.SSHCertPath(keysDir, proxyHost, tc.Config.Username, rootClusterName)
+
+	hostname := tc.Config.Host
+	port := tc.Config.HostPort
+	userHostString := hostname
+	login := ""
+	if tc.Config.HostLogin != "" {
+		login = tc.Config.HostLogin
+		userHostString = fmt.Sprintf("%v@%v", login, userHostString)
+	}
+
+	// format local command string (to run 'tsh proxy ssh')
+	localCommandString := formatLocalCommandString(cf.executablePath, rootClusterName)
+
+	keyName := fmt.Sprintf(`teleportHostCA-%v`, proxyHost)
+	if ok, err := addHostCAPublicKey(keyName, hostCAPublicKey, hostname); !ok {
+		log.Infof("Failed to add host CA key: %T", err)
```

Comment:
I'd maybe argue for `log.Errorf()` here:
```suggestion
		log.Errorf("Failed to add host CA key: %T", err)
```